lighthouse-core/scripts/legacy-javascript/run.js

```diff
@@ -0,0 +1,313 @@
+  'es6.reflect.set-prototype-of',
+  'es6.reflect.set',
+  'es6.set',
+  'es6.string.code-point-at',
+  'es6.string.ends-with',
+  'es6.string.from-code-point',
+  'es6.string.includes',
+  'es6.string.iterator',
+  'es6.string.raw',
+  'es6.string.repeat',
+  'es6.string.starts-with',
+  'es6.string.trim',
+  'es6.typed.array-buffer',
+  'es6.typed.data-view',
+  'es6.typed.float32-array',
+  'es6.typed.float64-array',
+  'es6.typed.int16-array',
+  'es6.typed.int32-array',
+  'es6.typed.int8-array',
+  'es6.typed.uint16-array',
+  'es6.typed.uint32-array',
+  'es6.typed.uint8-array',
+  'es6.typed.uint8-clamped-array',
+  'es6.weak-map',
+  'es6.weak-set',
+  'es7.array.includes',
+  'es7.object.entries',
+  'es7.object.get-own-property-descriptors',
+  'es7.object.values',
+  'es7.string.pad-end',
+  'es7.string.pad-start',
+];
+
+/**
+ * @param {{group: string, name: string, code: string, babelrc?: *}} options
+ */
+async function createVariant(options) {
+  const {group, name, code, babelrc} = options;
+  const dir = `${VARIANT_DIR}/${group}/${name.replace(/[^a-zA-Z0-9]+/g, '-')}`;
+
+  if (!fs.existsSync(`${dir}/main.bundle.js`) && (STAGE === 'build' || STAGE === 'all')) {
+    fs.mkdirSync(dir, {recursive: true});
+    fs.writeFileSync(`${dir}/main.js`, code);
+    fs.writeFileSync(`${dir}/.babelrc`, JSON.stringify(babelrc || {}, null, 2));
+    // Not used in this script, but useful for running Lighthouse manually.
+    // Just need to start a web server first.
+    fs.writeFileSync(`${dir}/index.html`, `<title>${name}</title><script src=main.bundle.min.js>`);
+
+    // Note: No babelrc will make babel a glorified `cp`.
+    execFileSync('yarn', [
+      'babel',
+      `${dir}/main.js`,
+      '--config-file', `${dir}/.babelrc`,
+      '--ignore', 'node_modules/**/*.js',
+      '-o', `${dir}/main.transpiled.js`,
+    ]);
+
+    // Transform any require statements (like for core-js) into a big bundle.
+    execFileSync('yarn', [
+      'browserify',
+      `${dir}/main.transpiled.js`,
+      '-o', `${dir}/main.bundle.js`,
+    ]);
+
+    // Minify.
+    execFileSync('yarn', [
+      'terser',
+      `${dir}/main.bundle.js`,
+      '-o', `${dir}/main.bundle.min.js`,
+    ]);
+  }
+
+  if (STAGE === 'audit' || STAGE === 'all') {
+    // Instead of running Lighthouse, use LegacyJavascript directly. Requires some setup.
+    // Much faster than running Lighthouse.
+    const documentUrl = 'http://localhost/index.html'; // These URLs don't matter.
+    const scriptUrl = 'https://localhost/main.bundle.min.js';
+    const networkRecords = [
+      {url: documentUrl},
+      {url: scriptUrl},
+    ];
+    const devtoolsLogs = networkRecordsToDevtoolsLog(networkRecords);
+    const jsRequestWillBeSentEvent = devtoolsLogs.find(e =>
+      e.method === 'Network.requestWillBeSent' && e.params.request.url === scriptUrl);
+    if (!jsRequestWillBeSentEvent) throw new Error('jsRequestWillBeSentEvent is undefined');
+    // @ts-ignore
```

Comment:
what's being ignored here... wouldn't the line about make us safe?

Reply:
the log event is not narrowed from `find` :/

Reply:
ah gotcha 👍 

```suggestion
    // @ts-ignore - the log event is not narrowed to 'Network.requestWillBeSent' even from find
```